### ISSUE TITLE
Bug with screensaver

### ISSUE THREAD

**Issue by artemstorozhuk** [koreader/kindlepdfviewer#85]
Hi,

I found interesting bug accosiated with showing screensaver:
1) Rotate image on 90 degrees;
2) Move slider to go to the sleep mode.

After doing this I see something like this:
![After doing this I see something like this.](http://img638.imageshack.us/img638/7183/img20120331112128.jpg)

And after moving slider again:
![And after moving slider again.](http://img36.imageshack.us/img36/8833/img20120331112152.jpg)

Original screensaver:
![Original screensaver.](http://img402.imageshack.us/img402/2471/00009anatomy.png)

**Comment by houqp** [comment]
Thanks for the report. You may want to look at this reply for some more explanation:
http://www.mobileread.com/forums/showpost.php?p=2016711&postcount=175

In short, current version of kindlepdfviewer does not support screen saver yet. We will fix this in next release. For now, if you want to go to screen saver, please remember to rotate the screen back ;P.

**Comment by houqp** [comment]
Ok, following @hwhw 's instruction, I found that this fix only works well under portrait mode, when we enter kindlepdfviewer in landscape mode, sometimes I get a blank screen in screen saver. @hwhw even reports that he can reproduce the cluttered screen in this way (I only managed to reproduce it once ;P ).

I guess this is caused by the race between the framework and kindlepdfviewer on power slider event occur. If the framework responses faster and try to draw on the screen before kindlepdfviewer rotates the screen back, things will go mess.

So I gave a try to fall back to our previous solution: stopping the framework. It turns out that the blank screen went away and so far I haven't encountered the cluttered screen. Stopping the framework also means we don't need to wait for 3 seconds after resume so it is much faster on resume. I think we might need to stop the framework again. :(

Any thoughts on fixing this bug?
